LucasNumber/LucasNumber/lucas.py

```diff
@@ -0,0 +1,44 @@
+"""Lucas Number Generator.(EE7:65)."""
+from collections import deque
+import numpy as np
+
+
+def golden_ratio_calc(sequence):
+    """
+    Compute the golden ration based on the current sequence points.
+
+    Parameters
+    ----------
+    sequence: list[int]
+        Two most recent values from Lucas Number sequence.
+
+    Return: float
+        Calculated golden ratio
+    """
+    if np.max(sequence[0]) > np.max(sequence[1]):
+        max_value = sequence[0]
+    else:
+        max_value = sequence[1]
+    return (sequence[0] + sequence[1]) / max_value
+
+
+def lucas_number_generator(seed):
+    """
+    Lucas number generator.
+
+    Parameters
+    ----------
+    seed: list[int]
+        Starting point for Lucas number generator. Two valid values required (e.g. [2,1])
+
+    Return: list[int, float]
+        Yields next Lucas number in sequence with current golden ratio
+    """
+    sequence = deque(seed)
+    while True:
+        lucas_number = sequence[0] + sequence[1]
+        golden_ratio = golden_ratio_calc(sequence)
+        next_ln = yield [lucas_number, golden_ratio]
```

Comment:
Nice usage of the co-routine idea.

You may just want to include in your function documentation that the user needs to send the next number back into the coroutine, and close when they are finished, because this isn't obvious from your function documentation.

The only reservation that I have, and this is more of a nit-pick (or requirements interpretation than anything else) is that in this case, generating the right sequence involves the user's code `send`ing the correct number back into the co-routine.

So there's nothing wrong with the code, in fact this may be exactly what is needed in some hypothetical situations, it's just not quite what I had in mind. This is more of a requirements analysis issue than anything else.

Reply:
Noted. The intention was to try use both features (yield, send) more than trying to produce a production piece of code.

Reply:
Yes, and as such this worked well. Hence why I said more of a nit-pick. I wouldn't suggest making any changes here, you've got the general generator concept quite well.